lib/livebook/evaluator.ex

```diff
@@ -155,7 +161,11 @@ defmodule Livebook.Evaluator do
     Evaluator.IOProxy.flush(state.io_proxy)
     Evaluator.IOProxy.clear_input_buffers(state.io_proxy)
 
-    output = state.formatter.format_response(response)
+    output = %{
+      response: state.formatter.format_response(response),
+      evaluation_time_ms: result_context.evaluation_time_ms
+    }
+
     send(send_to, {:evaluation_response, ref, output})
```

Comment:
We should leave `output` as is and just add a time entry to the tuple. To make this easier to extend in the future, the new element could be a map, something like this:

```elixir
metadata = %{evaluation_time_ms: evaluation_time_ms}
send(send_to, {:evaluation_response, ref, output, metadata})
```

@josevalim please double-check if this less strict policy makes sense

Reply:
So it will be good to add metadata to all messages 

```
+    * `{:evaluation_output, ref, output, metadata}` - output captured during evaluation
+    * `{:evaluation_response, ref, output, metadata}` - final result of the evaluation
```

Reply:
Cannot think of a use case for output metadata, but sounds like a good call to me yeah. Also please add a note to the second item like "recognised metadata entries are: `evaluation_time_ms`".

Reply:
Won't we have to change Kino if we add metadata to the output? 🤔

Reply:
Nah, Kino sends `{:livebook_put_output, output}` to the group leader, but the actual message is composed by `IOProxy`:

https://github.com/elixir-nx/livebook/blob/958c03b1953732e0012c3eed4694e1319fe7f104/lib/livebook/evaluator/io_proxy.ex#L191

https://github.com/elixir-nx/livebook/blob/958c03b1953732e0012c3eed4694e1319fe7f104/lib/livebook/evaluator/io_proxy.ex#L300